slack_sdk/models/blocks/block_elements.py

```diff
@@ -1381,6 +1405,185 @@ def __init__(
+
+
+class UrlInputElement(InputInteractiveElement):
+    type = "url_text_input"
+
+    @property
+    def attributes(self) -> Set[str]:
+        return super().attributes.union(
+            {
+                "initial_value",
+                "dispatch_action_config",
+            }
+        )
+
+    def __init__(
+        self,
+        *,
+        action_id: Optional[str] = None,
+        initial_value: Optional[str] = None,
+        dispatch_action_config: Optional[Union[dict, DispatchActionConfig]] = None,
+        focus_on_load: Optional[bool] = None,
+        placeholder: Optional[Union[str, dict, TextObject]] = None,
+        **others: dict,
+    ):
+        """
+        A URL input element, similar to the Plain-text input element,
+        creates a single line field where a user can enter URL-encoded data.
+        https://api.slack.com/reference/block-kit/block-elements#url
+
+        Args:
+            action_id (required): An identifier for the input value when the parent modal is submitted.
+                You can use this when you receive a view_submission payload to identify the value of the input element.
+                Should be unique among all other action_ids in the containing block.
+                Maximum length for this field is 255 characters.
+            initial_value: The initial value in the URL input when it is loaded.
+            dispatch_action_config: A dispatch configuration object that determines when during text input
+                the element returns a block_actions payload.
+            focus_on_load: Indicates whether the element will be set to auto focus within the view object.
+                Only one element can be set to true. Defaults to false.
+            placeholder: A plain_text only text object that defines the placeholder text shown in the URL input.
+                Maximum length for the text in this field is 150 characters.
+        """
+        super().__init__(
+            type=self.type,
+            action_id=action_id,
+            placeholder=TextObject.parse(placeholder, PlainTextObject.type),
+            focus_on_load=focus_on_load,
+        )
+        show_unknown_key_warning(self, others)
+
+        self.initial_value = initial_value
+        self.dispatch_action_config = dispatch_action_config
+
+
+# -------------------------------------------------
+# Number Input Element
+# -------------------------------------------------
+
+
+class NumberInputElement(InputInteractiveElement):
+    type = "number_input"
+
+    @property
+    def attributes(self) -> Set[str]:
+        return super().attributes.union(
+            {
+                "initial_value",
+                "is_decimal_allowed",
+                "min_value",
+                "max_value",
+                "dispatch_action_config",
+            }
+        )
+
+    def __init__(
+        self,
+        *,
+        action_id: Optional[str] = None,
+        is_decimal_allowed: Optional[bool] = None,
```

Comment:
I believe this is a required property

Reply:
Yess it is required in the docs but the python implementation of other `required fields` was not done, so I followed the trend of the existing code.

What do you think?

Reply:
Hmm, when I look at, for example, the Button element's `text` property, that is required, and is not marked as Optional, so that is why I asked / pointed it out. I am also confused 😄

Reply:
In this file it seems like all `action_id` are optional but should be required, not sure what the standard is here @seratch what do you think?

Reply:
The `action_id` one, I think, is a unique situation, as _technically_ (you can try it out yourself) it is not required. You can create a block kit block with elements without action_id, and it is fine. However, you _should_ provide it. So our public facing documentation says it is required, while in reality, it is not really required.

Reply:
@filmaj @WilliamBergamin First off, is_decimal_allowed is not required actually. So, I am pretty sure that the document is wrong.

>action_id

Fil's understanding is correct. For most real-world use cases, the property can be necessary but there is a rare case such as "We have only one interactive block in the app! so we'd like to skip setting action_id to distinguish the button." or something like that. So the platform does not technically require it.